src/metrics/mod.rs

```diff
@@ -3,13 +3,10 @@
 
 #![allow(clippy::all)]
 
-/// Counter implementation
-pub mod counter;
-/// Exponentially Weighted Moving Average Measurement
-pub mod ewma;
-/// Gauge implementation
-pub mod gauge;
-/// Histogram implementation
-pub mod histogram;
-/// Meter implementation
-pub mod meter;
+mod ewma;
+mod meter;
+#[cfg(all(feature = "hardware_counters", target_os = "linux"))]
+pub mod perf_event;
+
+#[cfg(feature = "metrics")]
```

Comment:
the ``feature = metrics`` part won't be needed here as the metrics module should be covered under the feature in src/lib.rs